File: ToWer.DbWrapper.VB/IDbWrapper.vb


Public Interface IDbWrapper

    Function ReadSingle(Of T As Class)(ByVal connectionString As String, ByVal procedureName As String) As T

    Function ReadSingle(Of T As Class)(ByVal connectionString As String, ByVal procedureName As String, ByVal parameters As Dictionary(Of String, Object)) As T

    Function ReadList(Of T As Class)(ByVal connectionString As String, ByVal procedureName As String) As List(Of T)

    Function ReadList(Of T As Class)(ByVal connectionString As String, ByVal procedureName As String, ByVal parameters As Dictionary(Of String, Object)) As List(Of T)

    Sub ExecuteNonQuery(ByVal connectionString As String, ByVal procedureName As String, ByVal parameters As Dictionary(Of String, Object))

    Sub ExecuteNonQuery(Of T As Class)(ByVal connectionString As String, ByVal procedureName As String, ByVal item As T)

End Interface

File: ToWer.DbWrapper/IDbWrapper.cs
using System.Collections.Generic;
using System.Data.SqlClient;

namespace ToWer.DbWrapper
{
    public interface IDbWrapper
    {
        void SetConnectionString(string connectionString);
        void SetStandardCommandParameters(List<SqlParameter> parameters);

        T ReadSingle<T>(string procedureName) where T : class;
        T ReadSingle<T>(string procedureName, Dictionary<string, object> parameters) where T : class;
        T ReadSingle<T>(string procedureName, List<SqlParameter> parameters) where T : class;
        T ReadSingle<T>(string connectionString, string procedureName) where T : class;
        T ReadSingle<T>(string connectionString, string procedureName, Dictionary<string, object> parameters) where T : class;
        T ReadSingle<T>(string connectionString, string procedureName, List<SqlParameter> parameters) where T : class;

        List<T> ReadList<T>(string procedureName) where T : class;
        List<T> ReadList<T>(string procedureName, Dictionary<string, object> parameters) where T : class;
        List<T> ReadList<T>(string procedureName, List<SqlParameter> parameters) where T : class;
        List<T> ReadList<T>(string connectionString, string procedureName) where T : class;
        List<T> ReadList<T>(string connectionString, string procedureName, Dictionary<string, object> parameters) where T : class;
        List<T> ReadList<T>(string connectionString, string procedureName, List<SqlParameter> parameters) where T : class;

        void ExecuteNonQuery(string procedureName, Dictionary<string, object> parameters);
        void ExecuteNonQuery(string procedureName, List<SqlParameter> parameters);
        void ExecuteNonQuery(string connectionString, string procedureName, Dictionary<string, object> parameters);
        void ExecuteNonQuery(string connectionString, string procedureName, List<SqlParameter> parameters);
        void ExecuteNonQuery<T>(string connectionString, string procedureName, T item) where T : class;
    }
}
